rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,140 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.manifests.kustomize;
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.rosco.jobs.BakeRecipe;
+import com.netflix.spinnaker.rosco.manifests.TemplateUtils;
+import com.netflix.spinnaker.rosco.manifests.kustomize.mapping.Kustomization;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Component
+@Slf4j
+public class KustomizeTemplateUtils extends TemplateUtils {
+
+    private HashSet<String> filesToDownload = new HashSet<String>();
+
+    @Autowired
+    private KustomizationFileReader kustomizationFileReader;
+
+    public BakeRecipe buildBakeRecipe(BakeManifestEnvironment env, KustomizeBakeManifestRequest request) {
+        BakeRecipe result = new BakeRecipe();
+        result.setName(request.getOutputName());
+        filesToDownload = new HashSet<>();
+        Artifact artifact = request.getInputArtifact();
+        if (artifact == null) {
+            throw new IllegalArgumentException("At least one input artifact must be provided to bake");
+        }
+        String kustomizationfilename = FilenameUtils.getName(artifact.getReference());
+        if (kustomizationfilename != null && !kustomizationfilename.toUpperCase().contains("KUSTOMIZATION")) {
+            throw new IllegalArgumentException("The inputArtifact should be a kustomization file valid.");
+        }
+
+        List<Artifact> artifacts = new ArrayList<>();
+        String templatePath = env.getStagingPath().resolve(FilenameUtils.getPath(artifact.getReference())).toString();
+        try {
+            Path path = Paths.get(FilenameUtils.getPath(artifact.getReference()));
+            artifact.setReference(path.toString());
+            HashSet<String> files = getFilesFromArtifact(artifact);
+            artifacts = files.stream()
+              .map(f -> {
+                return Artifact.builder()
+                    .reference(f)
+                    .artifactAccount(artifact.getArtifactAccount())
+                    .customKind(artifact.isCustomKind())
+                    .location(artifact.getLocation())
+                    .metadata(artifact.getMetadata())
+                    .name(artifact.getName())
+                    .provenance(artifact.getProvenance())
+                    .type(artifact.getType())
+                    .version(artifact.getVersion())
+                    .build();
+              }).collect(Collectors.toList());
+        } catch (IOException e) {
+            log.error("Error setting references in artifacts  " + e.getMessage());
```

Comment:
`log.error(..., e)` - log the exception

Reply:
Also the exception is logged but ignored. Shouldn't this stop the process?

Reply:
add `throw new IllegalStateException("Error setting references in artifacts " + e.getMessage(), e);`